beta/src/components/MDX/TerminalBlock.tsx

```diff
@@ -34,13 +35,36 @@ function TerminalBlock({level = 'info', children}: TerminalBlockProps) {
     message = children.props.children;
   }
 
+  const [copied, setCopied] = React.useState(false);
+  React.useEffect(() => {
```

Comment:
Trying to understand why we need a `useEffect` here? 
The state copied is going to be set as true only `onClick` right?

Can we execute something like this on click ? let me know your thoughts. 
```js
const copyText = () => {
    window.navigator.clipboard.writeText(message ?? '');
    setCopied(true);
    setTimeout(() => {
      setCopied(false);
    }, 2000);
  };

Reply:
My thinking was that because it is async, it should be treated as a side effect. I want to separate the logic of starting the side effect and the logic of the result of that side effect. Basically, the button responsibility is only triggering the side effect, and the side effect's responsibility is to update the DOM.

That being said, if you prefer, I am happy to rewrite this as just a function called onclick

Reply:
@harish-sethuraman would you prefer I switch it, or do you think that my reasoning is sound here and it should stay in a useEffect?

Reply:
I'm okay with leaving it as such. It took time for me to understand 😅.
we'll wait for other's review too.

PS: Thanks for the explanation.